src/core/Cline.ts

```diff
@@ -153,11 +154,32 @@ export class Cline {
 		// Initialize diffStrategy based on current state
 		this.updateDiffStrategy(Experiments.isEnabled(experiments ?? {}, EXPERIMENT_IDS.DIFF_STRATEGY))
 
-		if (task || images) {
-			this.startTask(task, images)
-		} else if (historyItem) {
-			this.resumeTaskFromHistory()
+		if (startTask) {
+			if (task || images) {
+				this.startTask(task, images)
+			} else if (historyItem) {
+				this.resumeTaskFromHistory()
+			} else {
+				throw new Error("Either historyItem or task/images must be provided")
+			}
+		}
+	}
+
+	static create(...args: ConstructorParameters<typeof Cline>): [Cline, Promise<void>] {
+		args[10] = false // startTask
+		const instance = new Cline(...args)
+
+		let task
+
+		if (args[6] || args[7]) {
+			task = instance.startTask(args[6], args[7])
+		} else if (args[8]) {
```

Comment:
These hardcoded indexes feel pretty fragile / easy for us to forget about next time we add a param to the constructor. Think there's anything we could do to keep them in sync? Maybe a comment at the very least?

Reply:
The "right" solution seems like using an actual type. I can do that as a follow-up right now; most of the `new Cline()` calls are in tests so it should be pretty safe.